source/lib/cdk-textract-stack.ts

```diff
@@ -233,20 +232,32 @@ export class CdkTextractStack extends cdk.Stack {
         {
           elasticsearchVersion: "6.5",
           elasticsearchClusterConfig: {
-            instanceType: "m5.large.elasticsearch"
+            instanceType: "m5.large.elasticsearch",
           },
           ebsOptions: {
             ebsEnabled: true,
             volumeSize: 20,
-            volumeType: "gp2"
+            volumeType: "gp2",
           },
           encryptionAtRestOptions: {
             enabled: true,
-            kmsKeyId: esEncryptionKey.keyId
-          }
+            kmsKeyId: esEncryptionKey.keyId,
+          },
         }
       );
     } else {
+      const serviceLinkedRole = new cdk.CfnResource(
+        this,
+        this.resourceName("es-service-linked-role"),
```

Comment:
Do we not need to give this role permissions for esLogGroup?

Reply:
Actually a service-linked role and its permissions already exist, in every AWS account, and they already include permissions between ES and log groups. The problem and resolution here was to actually link the service-linked role to the ES domain that we are creating. Otherwise, there would be the ES service-linked role, and an ES domain, but no communication between the two of them.